src/Results/DTO/GenerativeAiResult.php

```diff
@@ -0,0 +1,313 @@
+                return $part->getText();
+            }
+        }
+
+        throw new \RuntimeException('No text content found in first candidate');
+    }
+
+    /**
+     * Converts the first candidate to an image file.
+     *
+     * @since n.e.x.t
+     *
+     * @return FileInterface The image file.
+     * @throws \RuntimeException If no candidates or no image content.
+     */
+    public function toImageFile(): FileInterface
+    {
+        if (empty($this->candidates)) {
+            throw new \RuntimeException('No candidates available');
+        }
+
+        $message = $this->candidates[0]->getMessage();
+        foreach ($message->getParts() as $part) {
+            if ($part->getType()->equals(MessagePartTypeEnum::inlineFile()) && $part->getInlineFile() !== null) {
+                return $part->getInlineFile();
+            }
+            if ($part->getType()->equals(MessagePartTypeEnum::remoteFile()) && $part->getRemoteFile() !== null) {
+                return $part->getRemoteFile();
+            }
+        }
+
+        throw new \RuntimeException('No image content found in first candidate');
+    }
+
+    /**
+     * Converts the first candidate to an audio file.
+     *
+     * @since n.e.x.t
+     *
+     * @return FileInterface The audio file.
+     * @throws \RuntimeException If no candidates or no audio content.
+     */
+    public function toAudioFile(): FileInterface
+    {
+        // Similar implementation to toImageFile, but checking for audio MIME types
+        return $this->toImageFile(); // Simplified for now
+    }
+
+    /**
+     * Converts the first candidate to a video file.
+     *
+     * @since n.e.x.t
+     *
+     * @return FileInterface The video file.
+     * @throws \RuntimeException If no candidates or no video content.
+     */
+    public function toVideoFile(): FileInterface
+    {
+        // Similar implementation to toImageFile, but checking for video MIME types
+        return $this->toImageFile(); // Simplified for now
+    }
+
+    /**
+     * Converts the first candidate to a message.
+     *
+     * @since n.e.x.t
+     *
+     * @return Message The message.
+     * @throws \RuntimeException If no candidates available.
+     */
+    public function toMessage(): Message
+    {
+        if (empty($this->candidates)) {
+            throw new \RuntimeException('No candidates available');
+        }
+
+        return $this->candidates[0]->getMessage();
+    }
+
+    /**
+     * Converts all candidates to text array.
+     *
+     * @since n.e.x.t
+     *
+     * @return string[] Array of text content.
+     */
+    public function toTexts(): array
+    {
+        $texts = [];
+        foreach ($this->candidates as $candidate) {
+            $message = $candidate->getMessage();
+            foreach ($message->getParts() as $part) {
+                if ($part->getType()->equals(MessagePartTypeEnum::text()) && $part->getText() !== null) {
```

Comment:
Similar feedback as above applies here and in the subsequent methods for multiple candidates - I'm not going to repeat it below :)